src/p_interaction.cpp

```diff
@@ -112,28 +112,32 @@ void P_TouchSpecialThing (AActor *special, AActor *toucher)
 
 
 // [RH]
-// SexMessage: Replace parts of strings with gender-specific pronouns
+// ProMessage: Replace parts of strings with player-specific pronouns
 //
 // The following expansions are performed:
-//		%g -> he/she/it
-//		%h -> him/her/it
-//		%p -> his/her/its
+//		%g -> he/she/they/it
+//		%h -> him/her/them/it
+//		%p -> his/her/their/its
+//		%s -> his/hers/theirs/its
+//		%r -> he's/she's/they're/it's
 //		%o -> other (victim)
 //		%k -> killer
 //
-void SexMessage (const char *from, char *to, int gender, const char *victim, const char *killer)
+void ProMessage (const char *from, char *to, int pronoun, const char *victim, const char *killer)
 {
-	static const char *genderstuff[3][3] =
+	static const char *pronouns[GENDER_MAX][5] =
 	{
-		{ "he",  "him", "his" },
-		{ "she", "her", "her" },
-		{ "it",  "it",  "its" }
+		{ "he",   "him",  "his",   "his",    "he's"    },
+		{ "she",  "her",  "her",   "hers",   "she's"   },
+		{ "they", "them", "their", "theirs", "they're" },
+		{ "it",   "it",   "its",   "its'",   "it's"    }
 	};
-	static const int gendershift[3][3] =
+	static const int shift[GENDER_MAX][5] =
```

Comment:
I think this variable should be named something less ambiguous, i.e. "pronounshift"

Reply:
I find this to be fine, from the code's context it's very obvious what this does.